mlir/test/Conversion/MemRefToLLVM/invalid.mlir

```diff
@@ -22,7 +22,7 @@ func.func @bad_address_space(%a: memref<2xindex, "foo">) {
 
 // CHECK-LABEL: @invalid_int_conversion
 func.func @invalid_int_conversion() {
-     // expected-error@+1 {{conversion of memref memory space 1 : ui64 to integer address space failed. Consider adding memory space conversions.}}
+     // expected-error@unknown{{conversion of memref memory space 1 : ui64 to integer address space failed. Consider adding memory space conversions.}}
```

Comment:
(Non actionable)
This is a bit unfortunate regarding the quality of the error reporting, but is more of a pre-existing issue with the conversion functions (and the type converter infrastructure). I like the fact all type conversion related error handling is actually in the type conversions now.

Reply:
I'm thinking of maybe even removing this error entirely (not in this PR). This error is produced in the type converter. We don't usually produce errors there.

There are other ways to signal this to the user: when the type converter fails, the pattern does not apply. Either because a hand-written `converter.convertType` fails or because the values of the adaptor could not be computed. In either case, `-debug` would show that (`notifyMatchFailure`).

Reply:
That sounds great to me, I'd support that